client/src/main/java/io/pravega/client/stream/impl/EventStreamReaderImpl.java

```diff
@@ -175,7 +175,11 @@ private void releaseSegmentsIfNeeded() throws ReinitializationRequiredException
             SegmentInputStream reader = readers.stream().filter(r -> r.getSegmentId().equals(segment)).findAny().orElse(null);
             if (reader != null) {
                 if (groupState.releaseSegment(segment, reader.getOffset(), getLag())) {
-                    readers.remove(reader);
+                    try {
+                        readers.remove(reader);
+                    } finally {
+                        reader.close();
```

Comment:
Why is this in a finally block?
readers.remove is just removing an entry from a list. It should not be expected to throw.